runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ControlClientSink.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.fnexecution.control;
+
+/** A sink for {@link InstructionRequestHandler InstructionRequestHandlers} keyed by worker id. */
+@FunctionalInterface
+public interface ControlClientSink {
```

Comment:
<!--thread_id:cc_182927203_t; commit:fb7d2bd7ba1bb27186e8d2dda861acd08666a0c1; resolved:1-->
<!--section:context-quote-->
> **angoenka** wrote:
> ControlClientSink => ControlClientSinkPool
> and if it is a pool then we can provide ThrowingConsumer<InstructionRequestHandler> in get method.
> Usage will be like:
> `
>  ThrowingConsumer<InstructionRequestHandler> sink = controlClientSink.get(workerId);
> `

<!--section:body-->
It's not a sink pool per se, but rather a Sink that corresponds to a pool. I think moving this into a nested interface makes that relationship clearer.

I don't think it's great to have a new consumer be constructed on the fly for each incoming client. In this case, I think it makes sense to keep the Sink and Source interfaces as originally created and instead pass through the worker id.